egs/wham/TwoStep/model.py

```diff
@@ -0,0 +1,141 @@
+import os
+import asteroid.filterbanks.simple_adaptive as fb
+from asteroid.masknn.blocks import TwoStepTDCN
+from asteroid.engine.optimizers import make_optimizer
+from asteroid import torch_utils
+
+import torch
+import torch.nn as nn
+
+
+class Model(nn.Module):
+    def __init__(self, pretrained_filterbank, conf):
+        super().__init__()
+        self.pretrained_filterbank = pretrained_filterbank
+        self.separator = TwoStepTDCN(
+            pretrained_filterbank,
+            bn_chan=conf['masknet']['bn_chan'],
+            hid_chan=conf['masknet']['hid_chan'],
+            kernel_size=conf['masknet']['conv_kernel_size'],
+            n_blocks=conf['masknet']['n_blocks'],
+            n_repeats=conf['masknet']['n_repeats'],
+            n_sources=conf['masknet']['n_src'])
+
+    def get_ideal_targets(self, mixture, clean_sources):
+        """
+        Get the latent targets for all sources
+        :param mixture: Input mixture in time domain [batch, timesamples]
+        :param clean_sources: Clean sources that constitute to the mixture in
+            time domain [batch, n_sources, timesamples].
+        :return: Latent representations for the sources which can be used as
+            targets for training:
+            [batch, n_sources, timesamples//encoder_stride]
+        """
+        return self.pretrained_filterbank.get_encoded_sources(mixture,
+                                                              clean_sources)
+
+    def estimate_latent_representations(self, mixture):
+        return self.separator(mixture.unsqueeze(1))
+
+    def get_ideal_latent_targets(self, mixture, clean_sources):
+        return self.pretrained_filterbank.get_encoded_sources(mixture,
+                                                              clean_sources)
+
+    def forward(self, x):
+        # Infer sources in the time domain
+        return self.separator.infer_source_signals(x.unsqueeze(1))
+
+
+def get_encoded_paths(conf, train_part=None):
+    exp_dir = conf['main_args']['exp_dir']
+    N = conf['filterbank']['n_filters']
+    L = conf['filterbank']['kernel_size']
+    checkpoint_dir = os.path.join(
+        exp_dir, train_part, 'N_{}_L_{}_checkpoints/'.format(N, L))
+    return exp_dir, checkpoint_dir
```

Comment:
If several _different_ models are trained, they should have different `tag` from bash, and correspond to different experiment folders rather than having different checkpoint folders IMO.